kratos/tests/utilities/test_geometrical_projection_utilities.cpp

```diff
@@ -29,9 +29,9 @@ using GeometryType = typename GeometricalProjectionUtilities::GeometryType;
 GeometryType::Pointer CreateTriangle2D3NForTest()
 {
     GeometryType::PointsArrayType points;
-    points.push_back(Kratos::make_shared<NodeType>(1,0.04, 0.02, 0.0));
-    points.push_back(Kratos::make_shared<NodeType>(2,1.1, 0.03, 0.0));
-    points.push_back(Kratos::make_shared<NodeType>(3,1.08, 1.0, 0.0));
+    points.push_back(std::make_intrusive<NodeType>(1,0.04, 0.02, 0.0));
```

Comment:
Wouldn’t it be better to have Kratos::make_intrusive?

This way you could switch with one like on the intrusive ptr is in the standard 
Otherwise it might be a similar struggle as with boost::shared_prr

Reply:
@philbucher yes, your proposal makes sense. Actually i wanted to have Kratos::intrusive_ptr instead of std::shared_ptr. I'll take a look into this too

Reply:
It is undefined behavior to add new definitions (other than some tightly restricted template specializations) in the `std` namespace. It is also undefined behavior to use identifiers that begin with an underscore followed by an uppercase letter or contain two consecutive underscores. The original implementation of `intrusive_ptr` was uglified in order to win favor of maintainers of C++ standard library maintainers, but such uglification is not appropriate elsewhere. I suggest you reformat the original code and move it out of the `std` namespace.

Reply:
@lhmouse we have a "using" defined in our namespace for the std::shared_ptr

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/shared_pointers.h#L26

(and just below for the make_shared)

@philbucher suggested using the same mechanism for the std::intrusive_ptr.

Ideally our idea is to proceed in steps, and first substitute shared_ptr -> intrusive_ptr (your version, WITH weak_ptrs) so that changes are minimal.

then as a second step get rid of weak_ptrs (substituting by raw_ptrs) without having to change again.

This way we hide the "implementation details" of the intrusive_ptrs from our code base.
I guess you are way more experienced than me on this ... so if you see issues with this path forward please comment

Reply:
I actually tried to port your code to be in the Kratos namespace ... but failed to compile ;-)

however since it is in a sense it is "temporary" while we transition to a version without shared_ptrs (like the one you pointed me to) i think that the "using" version will do during the transition process (which will last quite a while...)

thank you for feedback!!